CondTools/Hcal/test/pfcuts_db_io_test.csh

```diff
@@ -0,0 +1,105 @@
+
+process.load("Configuration.StandardSequences.GeometryDB_cff")
+process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+from Configuration.AlCa.autoCond import autoCond
+process.GlobalTag.globaltag = autoCond["phase1_2022_realistic"] 
+
+process.load('Configuration.StandardSequences.Services_cff')
+process.load("CondCore.CondDB.CondDB_cfi")
+process.CondDB.connect = "sqlite_file:HcalPFCuts_V00_test.db"
+
+process.maxEvents = cms.untracked.PSet(
+    input = cms.untracked.int32(1)
+)
+
+process.source = cms.Source("EmptySource",
+    numberEventsInRun = cms.untracked.uint32(1),
+    firstRun = cms.untracked.uint32(1)
+)
+
+process.es_ascii = cms.ESSource("HcalTextCalibrations",
+   input = cms.VPSet(
+       cms.PSet(
+           object = cms.string("PFCuts"),
+           file = cms.FileInPath("CondTools/Hcal/data/hcalpfcuts.txt")
+      )
+   )
+)
+process.es_prefer = cms.ESPrefer('HcalTextCalibrations','es_ascii')
+
+process.PoolDBOutputService = cms.Service(
+    "PoolDBOutputService",
+    process.CondDB,
+    timetype = cms.untracked.string('runnumber'),
+    toPut = cms.VPSet(cms.PSet(
+        record = cms.string("HcalPFCutsRcd"),
+        tag = cms.string("HcalPFCuts_test_tag")
+    ))
+)
+
+process.mytest = cms.EDAnalyzer("HcalPFCutsPopConAnalyzer",
+    record = cms.string('HcalPFCutsRcd'),
+    loggingOn = cms.untracked.bool(True),
+    SinceAppendMode = cms.bool(True),
+    Source = cms.PSet(
+        IOVRun = cms.untracked.uint32(1)
+    )
+)
+process.p = cms.Path(process.mytest)
+%
+cmsRun temp_pfcuts_to_db.py
+rm temp_pfcuts_to_db.py
+
+cat >! temp_pfcuts_from_db.py <<%
+
+import FWCore.ParameterSet.Config as cms
+from Configuration.StandardSequences.Eras import eras
+
+process = cms.Process("FROMDB",eras.Run3)
+process.load("CondCore.CondDB.CondDB_cfi")
+
+process.load("Configuration.StandardSequences.GeometryDB_cff")
+process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+from Configuration.AlCa.autoCond import autoCond
+process.GlobalTag.globaltag = autoCond["phase1_2022_realistic"] 
+
+process.load('Configuration.StandardSequences.Services_cff')
+process.load("CondCore.CondDB.CondDB_cfi")
+process.CondDB.connect = "sqlite_file:HcalPFCuts_V00_test.db"
+
+process.maxEvents = cms.untracked.PSet(
+    input = cms.untracked.int32(1)
+)
+
+process.source = cms.Source("EmptySource",
+    numberEventsInRun = cms.untracked.uint32(1),
+    firstRun = cms.untracked.uint32(1)
+)
+
+process.PoolDBESSource = cms.ESSource("PoolDBESSource",
+    process.CondDB,
+    toGet = cms.VPSet(cms.PSet(
+        record = cms.string("HcalPFCutsRcd"),
+        tag = cms.string("HcalPFCuts_test_tag")
+    ))
+)
+
+process.dumpcond = cms.EDAnalyzer("HcalDumpConditions",
+    dump = cms.untracked.vstring("PFCuts")
+)
+process.p = cms.Path(process.dumpcond)
+%
+cmsRun temp_pfcuts_from_db.py
+rm temp_pfcuts_from_db.py
+
+diff DumpPFCuts_Run1.txt ${LOCAL_TOP_DIR}/src/CondTools/Hcal/data/hcalpfcuts.txt
```

Comment:
I don't think this is the correct syntax to address files in cms-data
@smuzaffar do you remember how to address it from the command line?

Reply:
I can see it in
```
$CMSSW_DATA_PATH/data-CondTools-Hcal/V00-01-00/CondTools/Hcal/data/hcalpfcuts.txt
```
but of course there must be some version independent way to access it...

Reply:
one can use
```
RefFile=$CMSSW_BASE/src/CondTools/Hcal/data/hcalpfcuts.txt
for dir in $(echo $CMSSW_SEARCH_PATH | tr ':' '\n') ; do
  [ -e $dir/CondTools/Hcal/data/hcalpfcuts.txt ] || continue
  RefFile=$dir/CondTools/Hcal/data/hcalpfcuts.txt 
  break
done
diff DumpPFCuts_Run1.txt  $RefFile
```

may be we can add a utility script to find a data file from CMSSW_SEARCH_PATH e.g `fileInPath.sh file` which then prints the full path of file and one can use it like
```
RefFile=$(fileInPath.sh CondTools/Hcal/data/hcalpfcuts.txt)
```